build/source/driver/multi_driver.f90

```diff
@@ -177,38 +216,105 @@ real(dp),parameter        :: safety = 0.01_dp               ! safety factor to e
 ! initialize the Jacobian flag
 doJacobian=.false.
 
+! allocate time structures
+call allocLocal(time_meta, refTime,   err=err, message=message); call handle_err(err,message)  ! reference time for the model simulation
+call allocLocal(time_meta, startTime, err=err, message=message); call handle_err(err,message)  ! start time for the model simulation 
+call allocLocal(time_meta, finshTime, err=err, message=message); call handle_err(err,message)  ! end time for the model simulation
+
 ! *****************************************************************************
-! (2) read model metadata
+! (2) populate/check metadata structures
 ! *****************************************************************************
-! initialize model metadata structures
-call init_metad(err,message); call handle_err(err,message)
 ! populate metadata for all model variables
 call popMetadat(err,message); call handle_err(err,message)
+! check data structures
+call checkStruc(err,message); call handle_err(err,message)
+
+! create the averageFlux metadata structure
+flux_mask = (flux_meta(:)%vartype == 'scalarv' .or. flux_meta(:)%vartype == 'ifcSoil')
```

Comment:
The mask is true if the variable is a scalar ​*OR*​ the variable is a flux at the layer interfaces. This is done because I want to calculate the mean flux of surface infiltration (at interface=0) and soil drainage (at interface=nSoil).